monai/networks/blocks/encoder.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABCMeta, abstractmethod
+from typing import Dict, List, Tuple
+
+__all__ = ["BasicEncoder"]
+
+
+class BasicEncoder(metaclass=ABCMeta):
+    """
+    Abstract class defines inferface of encoders in flexible unet.
```

Comment:
I think the class name could be `BaseEncoder`? 

typo "inferface" => "interface"